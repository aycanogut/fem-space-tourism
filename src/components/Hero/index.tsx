import { FC } from 'react'
import Transition from '../Transition'
import { IHeroProps } from '../../interfaces/Hero.interface'

const Hero: FC<IHeroProps> = ({ uppertitle, title, description }) => {
  return (
    <Transition
      initial={{ x: -200 }}
      animate={{ x: 0 }}
      transition={{ duration: 0.5 }}
    >
      <section className="mx-5 pt-28 text-center md:mx-0 md:pt-[200px] lg:pt-[300px] lg:text-left">
        <h2 className="font-barlow text-16 uppercase tracking-normal text-tropical_blue md:text-20 md:tracking-wider lg:text-28 lg:tracking-widest">
          {uppertitle}
        </h2>
        <h1 className="mt-1 font-bellefair text-80 font-normal uppercase text-white md:-mt-3 md:text-150 ">
          {title}
        </h1>
        <p className="text-15 mt-1 font-barlow leading-8 text-tropical_blue md:-mt-4 md:px-48 md:text-16 md:leading-9 lg:max-w-md lg:px-0 lg:pt-3 lg:text-18">
          {description}
        </p>
      </section>
    </Transition>
  )
}

export default Hero
