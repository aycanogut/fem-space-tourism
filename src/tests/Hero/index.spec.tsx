import renderer from 'react-test-renderer'
import Hero from '../../components/Hero'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Hero
        uppertitle="SO, YOU WANT TO TRAVEL TO"
        title="SPACE"
        description="Let’s face it; if you want to go to space, you might as well genuinely go to outer space and not hover kind of on the edge of it. Well sit back, and relax because we’ll give you a truly out of this world experience!"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
