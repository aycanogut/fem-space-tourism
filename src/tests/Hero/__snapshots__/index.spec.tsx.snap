// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="mx-5 pt-28 text-center md:mx-0 md:pt-[200px] lg:pt-[300px] lg:text-left"
>
  <h2
    className="font-barlow text-16 uppercase tracking-normal text-tropical_blue md:text-20 md:tracking-wider lg:text-28 lg:tracking-widest"
  >
    SO, YOU WANT TO TRAVEL TO
  </h2>
  <h1
    className="mt-1 font-bellefair text-80 font-normal uppercase text-white md:-mt-3 md:text-150 "
  >
    SPACE
  </h1>
  <p
    className="text-15 mt-1 font-barlow leading-8 text-tropical_blue md:-mt-4 md:px-48 md:text-16 md:leading-9 lg:max-w-md lg:px-0 lg:pt-3 lg:text-18"
  >
    Let’s face it; if you want to go to space, you might as well genuinely go to outer space and not hover kind of on the edge of it. Well sit back, and relax because we’ll give you a truly out of this world experience!
  </p>
</section>
`;
